lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -217,6 +208,60 @@ def el8?
       def el_major_version
         return os_facts['os']['release']['major'] if el?
       end
+
+      def foreman_plugin_prefix
```

Comment:
I wonder if the following is more "plugable"

```ruby
def ruby_prefix(scl=true)
  if el7? && scl
    'tfm-rubygem-'
  elsif el7? || el8?
    'rubygem-'
  elsif debian?
    'ruby-'
  end
end

def foreman_plugin_name(plugin)
  if debian?
    plugin.tr!('_', '-')
  end
  ruby_prefix + plugin
end

def proxy_plugin_name(plugin)
  if debian?
    plugin.tr!('_', '-')
    proxy_plugin_prefix = 'smart-proxy-'
  else
    proxy_plugin_prefix = 'smart_proxy_'
  end
  scl = foreman_version >= '2.0'
  ruby_prefix(scl) + proxy_plugin_prefix + plugin
end
```

I intentionally left out the 'foreman' out of `foreman_plugin_name`, as we have at least one plugin that is called `foreman-tasks` (and not `foreman_tasks`)

Not insisting on the above, just giving ideas.